spec/stripe_tester_spec.rb

```diff
@@ -97,6 +115,38 @@
       expect(response).to be_truthy
     end
 
+    it "#post_to_url should return true when authentication is provided" do
+      data = StripeTester.load_template(:invoice_created)
+      url = "http://localhost:3000/transactions"
+      fake_web_url = "http://stripe:password@localhost:3000/transactions"
+      StripeTester.webhook_url = url
+      StripeTester.webhook_password='password'
+
+      FakeWeb.register_uri(:post,
+                           fake_web_url,
+                           body: data.to_json,
+                           content_type: 'application/json')
+
+      response = StripeTester.post_to_url(data)
+
+      expect(response).to be_truthy
+    end
+
+    it "#post_to_url should return true when authentication is provided through url" do
+      data = StripeTester.load_template(:invoice_created)
+      url = "http://stripe:password@localhost:3000/transactions"
+      StripeTester.webhook_url = url
+
+      FakeWeb.register_uri(:post,
+                           url,
+                           body: data.to_json,
+                           content_type: 'application/json')
+
+      response = StripeTester.post_to_url(data)
+
+      expect(response).to be_truthy
+    end
+
```

Comment:
Could you also add a test to assert that the request fails if the URL has authentication but the user has NOT provided a `webhook_password`. 

Reference: https://github.com/chrisk/fakeweb#using-http-basic-authentication